<!--
 Half-ring of IR sensors
  - base to ring center static tf
  - IR readings to laserscan node

 To interface with waiterbot's Arduino Mega 2560 board, flash the firmware provided on Bosch drivers:

  $ roscd arduino_interface
  $ cd arduino_firmware
  $ export BOARD=mega2560
  $ export ARDUINO_DIR=/opt/arduino  # (or whatever you installed arduino software)
  $ make upload

 You must install the Arduino udev rule to make the board available on the default port "/dev/arduino":

  $ roscd waiterbot_sensors
  $ sudo cp resources/ir_scan/58-arduino.rules /etc/udev/rules.d
 
 and then replug the USB cable.
-->
<launch>
  <arg name="base_link" default="base_link" />
  <arg name="irscan_topic" default="ir_scan"/>

  <node pkg="tf" type="static_transform_publisher" name="ir_link_publisher"
        args="0.0 0.0 0.09 0.0 0.0 0.0 $(arg base_link) /ir_link 100"/>

  <node name="ir_to_laserscan" pkg="waiterbot_sensors" type="waiter_ir_scan_node">
    <param name="arduino_port" value="/dev/arduino"/>
    <param name="maximum_range" value="0.65"/>
    <param name="rangers_count" value="7"/>
    <param name="infinity_range" value="6.0"/>
    <param name="range_variance" value="0.025"/>
    <remap from="ir_scan" to="$(arg irscan_topic)"/>
  </node>
</launch>
